mmv1/third_party/terraform/.teamcity/components/builds/build_steps.kt

```diff
@@ -226,3 +226,54 @@ fun BuildSteps.archiveArtifactsIfOverLimit() {
         // https://youtrack.jetbrains.com/issue/KT-2425/Provide-a-way-for-escaping-the-dollar-sign-symbol-in-multiline-strings-and-string-templates
     })
 }
+
+fun BuildSteps.runDiffTests() {
+    if (UseTeamCityGoTest) {
+        step(ScriptBuildStep {
+            name = "Run Tests"
+            scriptContent = "go test -v \"%PACKAGE_PATH%\" -timeout=\"%TIMEOUT%h\" -test.parallel=\"%PARALLELISM%\" -run=\"%TEST_PREFIX%\" -json"
+        })
+    } else {
+        step(ScriptBuildStep {
+            name = "Compile Test Binary"
+            workingDir = "%PACKAGE_PATH%"
+            scriptContent = """
+                #!/bin/bash
+                export TEST_FILE_COUNT=$(ls ./*_test.go | wc -l)
+                if test ${'$'}TEST_FILE_COUNT -gt "0"; then
+                    echo "Compiling test binary"
+                    go test -c -o test-binary
+                else
+                    echo "Skipping compilation of test binary; no Go test files found"
+                fi
+            """.trimIndent()
+        })
+
+        step(ScriptBuildStep {
+            name = "Run via jen20/teamcity-go-test"
```

Comment:
Could you change the name to reflect the new behavior? We aren't using jen20/* as far as I know